lib/state_machine.js

```diff
@@ -27,99 +27,196 @@ limitations under the License.
     module.exports = factory();
   }
 }(function() {
```

Comment:
One day, but not until browsers actually implement ES6, or we revert to compiling everything and losing fast edit-refresh capabilities. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import